frontend/lib/post.ts

```diff
@@ -0,0 +1,9 @@
+import { readdir, readFile } from "node:fs/promises";
+import { join } from "node:path";
+
+export async function readPosts(): Promise<string[]> {
+  const filenames = await readdir("posts");
+  return Promise.all(
+    filenames.map((filename) => readFile(join("posts", filename), "utf8"))
+  );
```

Comment:
並びがreaddirに依存しているが、あるべき姿としては公開日順など制御可能な状態である必要があると思った

```suggestion
  // TODO: 公開日順など並びが制御可能な状態にして
  const filenames = await readdir("posts");
  return Promise.all(
    filenames.map((filename) => readFile(join("posts", filename), "utf8"))
  );
```